framework/include/utils/MathUtils.h

```diff
@@ -316,4 +316,26 @@ mooseSetToZero(std::vector<Real> & vec)
     v = 0.;
 }
 
+/**
+ * generate a complete multi index table for given dimension and order
+ * i.e. given dim = 2, order = 2, generated table will have the following content
+ * 0 0
+ * 1 0
+ * 0 1
+ * 2 0
+ * 1 1
+ * 0 2
+ * The first number in each entry represents the order of the first variable, i.e. x;
+ * The second number in each entry represents the order of the second variable, i.e. y.
+ * Multiplication is implied between numbers in each entry, i.e. 1 1 represents x^1 * y^1
+ *
+ * @param dim dimension of the multi-index, here dim = mesh dimension
+ * @param order generate the multi-index up to certain order
+ * @return a data structure holding entries representing the complete multi index
+ */
+std::vector<std::vector<unsigned int>> multiIndex(unsigned int dim, unsigned int order);
```

Comment:
Order here is a polynomial order, right? When I first read `multiIndex` I thought it was an unrolled way to iterate over rank N tensor entries.

Reply:
It could be a polynomial order. This method gives all multi indices up to order `order`. Say we have a multi index `a`, then this method gives all multi indices with `|a| <= order`. Perhaps there's a better name, but I think the description is clear enough.